openml/datasets/dataset.py

```diff
@@ -561,12 +520,12 @@ def _load_data(self):
         fpath = self.data_feather_file if self.cache_format == "feather" else self.data_pickle_file
         logger.info(f"{self.cache_format} load data {self.name}")
         try:
-            if self.cache_format == "feather":
+            if self.cache_format == "feather" and self.feather_attribute_file:
                 data = pd.read_feather(self.data_feather_file)
                 fpath = self.feather_attribute_file
                 with open(self.feather_attribute_file, "rb") as fh:
                     categorical, attribute_names = pickle.load(fh)
-            else:
+            elif self.data_pickle_file:
```

Comment:
Could you please add an `else` again to make sure we're not running into some edge case?